src/Reflection/Adapter/ReflectionClass.php

```diff
@@ -0,0 +1,433 @@
+    {
+        $properties = $this->betterReflectionClass->getProperties();
+
+        $wrappedProperties = [];
+        foreach ($properties as $key => $property) {
+            $wrappedProperties[$key] = new ReflectionProperty($property);
+        }
+        return $wrappedProperties;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function hasConstant($name)
+    {
+        return $this->betterReflectionClass->hasConstant($name);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getConstants()
+    {
+        return $this->betterReflectionClass->getConstants();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getConstant($name)
+    {
+        return $this->betterReflectionClass->getConstant($name);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getInterfaces()
+    {
+        $interfaces = $this->betterReflectionClass->getInterfaces();
+
+        $wrappedInterfaces = [];
+        foreach ($interfaces as $key => $interface) {
+            $wrappedInterfaces[$key] = new ReflectionClass($interface);
+        }
+        return $wrappedInterfaces;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getInterfaceNames()
+    {
+        return $this->betterReflectionClass->getInterfaceNames();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function isInterface()
+    {
+        return $this->betterReflectionClass->isInterface();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getTraits()
+    {
+        $traits = $this->betterReflectionClass->getTraits();
+
+        $wrappedTraits = [];
+        foreach ($traits as $key => $trait) {
+            $wrappedTraits[$key] = new ReflectionClass($trait);
+        }
+        return $wrappedTraits;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getTraitNames()
+    {
+        return $this->betterReflectionClass->getTraitNames();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getTraitAliases()
+    {
+        return $this->betterReflectionClass->getTraitAliases();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function isTrait()
+    {
+        return $this->betterReflectionClass->isTrait();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function isAbstract()
+    {
+        return $this->betterReflectionClass->isAbstract();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function isFinal()
+    {
+        return $this->betterReflectionClass->isFinal();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getModifiers()
+    {
+        return $this->betterReflectionClass->getModifiers();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function isInstance($object)
+    {
+        return $this->betterReflectionClass->isInstance($object);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function newInstance($args = null, $_ = null)
```

Comment:
@Ocramius it seems PHP7 complains about method signature not matching (see [build 347.2](https://travis-ci.org/Roave/BetterReflection/jobs/76068447#L241)). I've changed this back to avoid errors.